### PR TITLE
set a non-root user for containers

### LINKED ISSUE #2488: Dockerfile+Earthfile Docker Container User Not Set
### Share your bug report, feature request, or comment.

The Dockerfile provided with the Owncast code runs the application with user root

The `Dockerfile` does not specify or create a new user for running the application server, and by default if no user is specified docker will run the application as the root user inside that container.

The application should instead use a user with the minimum required privileges, to ensure that the application can only access to the strictly required functionalities.

Recommendation

Run the application as a user without root capabilities by modifying the `Dockerfile` document.

### PR DESCRIPTION
**Pull request by MFTabriz**
- set container user to "owncast"
- activate container builds on git push and pull requests (push only scheduled builds to container registry)
- update dockerfile based on earthfile (+docker)
- add dockerfile build and misconfiguration test to ci
- consolidate into a single nightly container build workflow and script

**Note:** deployment to container registry has not been tested!

### CODE REVIEW

File: .github/workflows/javascript-formatting.yml [hunk 1]
```diff
@@ -61,7 +61,6 @@ jobs:
           ref: ${{ github.event.pull_request.head.ref }}
           repository: ${{ github.event.pull_request.head.repo.full_name }}
           fetch-depth: 0
-          pull: --rebase --autostash
```
Comment (gabek) [inline .github/workflows/javascript-formatting.yml:64]: Why was this removed?

File: .github/workflows/container.yaml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      - name: Checkout repo
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Build and push
+        if: ${{ github.event_name == 'schedule' && env.GH_CR_PAT != null }}
+        env:
+          GH_CR_PAT: ${{ secrets.GH_CR_PAT }}
+          EARTHLY_BUILD_TAG: 'webv2'
```
Comment (gabek) [inline .github/workflows/container.yaml:49]: If this is going to be merged into `webv2` then the builds in `develop` are going to keep running. Those jobs won't get overwritten until `webv2` get merged into `develop`, and when that happens there will be no `webv2` branch anymore. So we should either plan for when that merge happens, or plan for now when both nightly and webv2 builds are happening. This PR kind of is in a middle ground.

File: .github/workflows/container-lint.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  trivy:
+    name: Dockerfile
+    runs-on: ubuntu-latest
+    container:
+      image: aquasec/trivy
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Check critical issues
+        run: trivy config --exit-code 1 --severity "HIGH,CRITICAL" ./Dockerfile
```
Comment (gabek) [inline .github/workflows/container-lint.yml:25]: What's trivy?

File: build/release/container.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+echo "Building container image ${EARTHLY_IMAGE_NAME}:${BUILD_TAG} ..."
+
+# Change to the root directory of the repository
+cd "$(git rev-parse --show-toplevel)" || exit
+if [ -n "${EARTHLY_BUILD_BRANCH}" ]; then
+    git checkout "${EARTHLY_BUILD_BRANCH}" || exit
+fi
+
+earthly --ci +docker-all --image="ghcr.io/owncast/${EARTHLY_IMAGE_NAME}" --tag="${BUILD_TAG}" --version="${VERSION}"
```
Comment (gabek) [inline build/release/container.sh]: Since you generalized this script to just `container.sh` it implies it's used to build all the container images, but that's not the case since Github is hard-coded here. Can the script maybe be renamed `nightlies.sh` or something to give a hint to people looking at it that it's not used for production builds and only used for nightly test builds?

File: .github/workflows/javascript-formatting.yml (hunk 1 above)
Comment (MFTabriz) [reply .github/workflows/javascript-formatting.yml:64]: For example [here](https://github.com/owncast/owncast/actions/runs/3791338784/jobs/6446694713#step:3:1):
image

File: .github/workflows/container-lint.yml (hunk 3 above)
Comment (MFTabriz) [reply .github/workflows/container-lint.yml:25]: I don’t have any specific preferences. Which dockerfile checker do you want to use?

File: .github/workflows/javascript-formatting.yml (hunk 1 above)
Comment (gabek) [reply .github/workflows/javascript-formatting.yml:64]: Ah ha! I remember that working somewhere from something before but I must have mixed it up with maybe a different action.

File: .github/workflows/container-lint.yml (hunk 3 above)
Comment (gabek) [reply .github/workflows/container-lint.yml:25]: I was just curious what trivy was. So it's a linter for Dockerfiles? Does it support Earthfiles too?

Comment (MFTabriz) [reply .github/workflows/container-lint.yml:25]: it is a security scanner https://aquasecurity.github.io/trivy/v0.35/ Other than dockerfiles, it supports many other formats but unfortunately it does not support Earthfiles yet.
It can detect if we use root as the user inside the container (the main issue addressed by this PR). It also had some recommendations for the dockerfile which I applied. See the report for non-critical issues [here]( https://github.com/owncast/owncast/actions/runs/3776695073/jobs/6420111585)

File: build/release/container.sh (hunk 4 above)
Comment (MFTabriz) [reply build/release/container.sh]: I renamed `build/release/container.sh` to `build/develop/container.sh`. I think this should make it clear enough. what do you think?

File: .github/workflows/container.yaml (hunk 2 above)
Comment (MFTabriz) [reply .github/workflows/container.yaml:49]: It is in the list for #2437
After the merge, all references to `webv2` in workflow files must be replaced with `develop`

File: build/release/container.sh (hunk 4 above)
Comment (gabek) [reply build/release/container.sh]: Maybe a comment on top would help. I just am afraid somebody will want to make changes to the release docker image and start editing this script and I'll have to tell them the script isn't even used, so I'd rather make it super clear.

Comment (gabek) [reply build/release/container.sh]: I have the same concern with the Dockerfile vs. Earthfile.

Comment (MFTabriz) [reply build/release/container.sh]: I added a note to the `Dockerfile`. Also made it clear in `build/develop/container.sh`.

File: .github/workflows/container.yaml (hunk 2 above)
Comment (gabek) [reply .github/workflows/container.yaml:49]: But before the merge these workflows won't even run, will they? Don't they need to be in the primary branch to run? Or that's what I assumed, I could be wrong I've never tried it. That's why I don't have `webv2` nightlies firing from the `webv2` branch.

Comment (MFTabriz) [reply .github/workflows/container.yaml:49]: These should get triggered for the PRs to `webv2` too. (as they have ran for this PR) Only the deployment to GH part is skipped.

PS: see the job named "Build development container / Earthly (pull_request) " below.
